echopype/echodata/combine.py

```diff
@@ -47,13 +47,17 @@ def check_zarr_path(
     """
 
     if zarr_path is not None:
-        # check that zarr_path is a string
-        if not isinstance(zarr_path, str):
-            raise TypeError(f"zarr_path must be of type {str}")
+        # ensure that zarr_path is a Path object, throw an error otherwise
+        if isinstance(zarr_path, str) or isinstance(zarr_path, Path):
+            zarr_path = Path(zarr_path)
+        else:
+            raise TypeError(
+                "The provided zarr_path input must be of type string " "or pathlib.Path!"
+            )
```

Comment:
```suggestion
        # ensure that zarr_path is a Path object, throw an error otherwise
        if not isinstance(zarr_path, (str, Path)):
            raise TypeError(
                "The provided zarr_path input must be of type string " "or pathlib.Path!"
            )
```